tests/unit/Remote/DesiredCapabilitiesTest.php

```diff
@@ -143,9 +143,9 @@ public function testShouldConvertCapabilitiesToW3cCompatible(
         DesiredCapabilities $inputJsonWireCapabilities,
         array $expectedW3cCapabilities
     ) {
-        $this->assertEquals(
-            $expectedW3cCapabilities,
-            $inputJsonWireCapabilities->toW3cCompatibleArray()
+        $this->assertJsonStringEqualsJsonString(
```

Comment:
I suggest adding testcase below, which will use of JsonSerializable (and which would cover the scenario from #849).

Something like:
```
            'chromeOptions as W3C capability should be converted' => [
                new DesiredCapabilities([
                    ChromeOptions::CAPABILITY_W3C => $chromeOptions,
                ]),
                [
                    'goog:chromeOptions' => new \ArrayObject(
                        [
                            'args' => ['--headless'],
                        ]
                    ),
                ],
            ],
```

Reply:
Done - this testcase sufficed with no need to make any other changes.